lib/grape/dsl/inside_route.rb

```diff
@@ -135,6 +135,7 @@ def version
       # @param message [String] The message to display.
       # @param status [Integer] the HTTP Status Code. Defaults to default_error_status, 500 if not set.
       def error!(message, status = nil, headers = nil)
+        headers = headers.present? ? header.merge(headers) : header
```

Comment:
There's a lot of `header` vs. `headers` going on here. I would rename `headers = nil` into `additional_headers` and use `headers` everywhere else in the body of this method as `header` and `headers` are aliases for semantic reasons: use `header` when accessing one and use `headers` when accessing all.